repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala

```diff
@@ -135,9 +162,47 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
     }
   }
 
+  private def handleJsonRequestLinks(wrappedRequest: HttpServletRequestWrapper, previousToken: Option[String], nextToken: Option[String], token: Option[String], applicableLinkPaths: List[LinkPath]): Try[HttpServletRequestWrapper] = {
+    val tryParsedJson = Try(Json.parse(wrappedRequest.getInputStream))
+    applicableLinkPaths.foldLeft(tryParsedJson)(transformJsonLink(_, _, token.get, previousToken, nextToken, isRequest = true)) match {
+      case Success(jsValue) =>
+        val jsValueBytes = jsValue.toString.getBytes(Charset.forName(Option(wrappedRequest.getCharacterEncoding).getOrElse("UTF-8")))
+        Success(new HttpServletRequestWrapper(wrappedRequest, new ServletInputStreamWrapper(new ByteArrayInputStream(jsValueBytes))))
+      case Failure(e) => Failure(e)
+    }
+  }
+
+  private def handleXmlRequestLinks(wrappedRequest: HttpServletRequestWrapper, previousToken: Option[String], nextToken: Option[String], token: Option[String], applicableLinkPaths: List[LinkPath]): Try[HttpServletRequestWrapper] = {
+    try {
+      val result = applicableLinkPaths.foldLeft(wrappedRequest.getInputStream) { (in: InputStream, linkPath: LinkPath) =>
+        val out = new ByteArrayOutputStream()
+        val transformer = templateMapRequest(linkPath).newTransformer
+        transformer.asInstanceOf[Controller].addLogErrorListener
+        transformer.setParameter("removedToken", token.getOrElse(""))
+        transformer.setParameter("prefixToken", previousToken.getOrElse(""))
```

Comment:
Man passing these around everywhere really feels like planning for a change we may never even do. If we redo this filter i think we are more likely to go with explicitly telling us where you want things versus this grey maybe it'll end up somewhere smart that it does right now.